docs/proposals/platform-extension.md

```diff
@@ -0,0 +1,105 @@
+# Platform Extension
+**Author**: Marcin Franczyk, Sascha Haase, Moritz Bracht, Sankalp Rangare
+
+**Status**: Draft proposal.
+
+## Goals
+
+This proposal is to enhance the current ADDON/extension mechanism and introduce a new platform-extension mechanism which will provide easier way to manage the lifecycle of extensions.
+
+## Motivation and Background
+### Differentiating between KKP AddOns and Platform Extensions
+Problem Statement: We want to be able to determine in early stages of development what new feature/s to push and how they can work out from a tech and business perspective. Therefore we gathered some early ideas on how to differentiate between two different yet similar  terms.
+
+Furthermore the following differentiators will allow us to determine if we should invest time & budget into a certain topic and if they intertwine. If a new topic has the potential to generate a standalone revenue stream, development efforts could be much higher and would result in a “platform extension”. There are requirements that we still need to fulfill (often called hygiene features in product management) the complexity and budget associated with it has to be reduced compared to the standalone feature and would result in the Extension mechanic.
+
+### Current Addon Mechanism
+#### Main Concept Points
+
+* Default Addons (core addons)
```

Comment:
I know I wrote this, but I would remove "(core addons)", core addons should indicate the addons prepared Kubermatic team IMO.